tools/codeformat.py

```diff
@@ -151,6 +151,11 @@ def main():
     cmd_parser.add_argument("-c", action="store_true", help="Format C code only")
     cmd_parser.add_argument("-p", action="store_true", help="Format Python code only")
     cmd_parser.add_argument("-v", action="store_true", help="Enable verbose output")
+    cmd_parser.add_argument(
+        "-f",
+        action="store_true",
+        help="Treat files argument as a list of names not globbing patterns, and apply default inclusion/exclusion filters",
```

Comment:
Just wondering: is this neeed? Expanding a glob which has no globbing specifier in it just yields the original string already? I mean, doesn't codeformat.py already work like this so you just pass complete paths and it does what is needed?

Reply:
It's needed to apply the include/exclude filter. If you do `codeformat.py somefile.c` then that file gets checked always - even if is not included the PATHS list, or if it's excluded in the EXCLUSIONS list (i.e. not checked in CI).

However for the pre-commit hook we only want to check a file if it would also be checked in CI.

I'll rename the help text to make it clearer, something like "If a list of files is given on the command line, filter the list against paths to include/exclude from the default CI check."

Reply:
Ok sorry I didn't read the rest of the code and wrongfully assumed it would always apply filtering, also because that seems to be the common way formatting tools work

Reply:
Yeah, no problems it's a bit fiddly. It is useful to bypass the filters sometimes, i.e. if you want to check if a file or directory can be added to the CI check. I've rewritten the help text a bit to try and make it clearer.

Reply:
For consistency I'd reword that to match the rest of the arguments. A bit hard to keep it short, but something like ''Filter files with the default exclusions (files must be actual paths not globs)". 

However I was thinking: why not use `files = list_files(args.files, EXCLUSIONS, TOP)` instead of listing files twice then filtering? That would still work with yourintended usage but also allow globs, making it much more useful, and no need to call list_files twice.

Reply:
> However I was thinking: why not use files = list_files(args.files, EXCLUSIONS, TOP)

That doesn't work, at least for the pre-commit use case, because there is an inclusions list as well (`PATHS` variable). pre-commit needs to only check files which are simultaneously included in `PATHS` and not excluded by `EXCLUSIONS`. But you're right it would be possible to call `list_files` twice and support globbing here, I've done that and tried to make the logic as concise as possible.

> For consistency I'd reword that to match the rest of the arguments.

I've shortened it a bit. It's still a bit longer than the others, but that's because there is actually some subtlety in the behaviour here and I think it's better to try and explain that to the user.